python-package/lightgbm/callback.py

```diff
@@ -2,14 +2,27 @@
 """Callbacks library."""
 import collections
 from operator import gt, lt
+from typing import Any, Callable, List, Union
 
 from .basic import _ConfigAliases, _log_info, _log_warning
 
 
+class CallbackWithAttributes():
```

Comment:
```suggestion
class _CallbackWithAttributes:
```

Could you please prefix this with `_`? Since this is just an internal detail for type-checking, I don't want to encourage `lightgbm` users to import this in their own code.